File: generate-pwa-icons.py
#!/usr/bin/env python3
"""
PWA Icon Generator
Generates all required PWA icons
"""

import os
import sys

try:
    from PIL import Image, ImageDraw, ImageFont
    PIL_AVAILABLE = True
except ImportError:
    PIL_AVAILABLE = False

icon_sizes = [72, 96, 128, 144, 152, 192, 384, 512]
output_dir = os.path.join(os.path.dirname(__file__), 'public', 'icons')
favicon_path = os.path.join(os.path.dirname(__file__), 'public', 'favicon.ico')

# Create icons directory
os.makedirs(output_dir, exist_ok=True)
print(f"Icons directory: {output_dir}")

if not PIL_AVAILABLE:
    print("ERROR: PIL/Pillow is not installed.")
    print("Install it with: pip3 install Pillow")
    print("\nAlternatively:")
    print("1. Use online tools: https://www.pwabuilder.com/imageGenerator")
    print("2. Install ImageMagick: sudo apt-get install imagemagick")
    print("3. Manually create icons and place them in:", output_dir)
    sys.exit(1)

# Try to load favicon
source_image = None
if os.path.exists(favicon_path):
    try:
        source_image = Image.open(favicon_path)
        print(f"Loaded favicon from: {favicon_path}")
    except Exception as e:
        print(f"Could not load favicon: {e}")
        source_image = None

# If no source image, create a simple one
if source_image is None:
    print("Creating simple E-Tinda icon...")
    # Create a 512x512 base image
    source_image = Image.new('RGBA', (512, 512), (40, 167, 69, 255))  # #28a745 green
    draw = ImageDraw.Draw(source_image)

    # Draw a white circle
    center = (256, 256)
    radius = 200
    draw.ellipse(
        [center[0] - radius, center[1] - radius, center[0] + radius, center[1] + radius],
        fill=(255, 255, 255, 255)
    )

    # Draw a simple leaf shape (E-Tinda logo)
    leaf_points = [
        (256, 156),  # top
        (336, 216),  # right
        (316, 296),  # right bottom
        (256, 336),  # bottom
        (196, 296),  # left bottom
        (176, 216),  # left
    ]
    draw.polygon(leaf_points, fill=(40, 167, 69, 255))

    # Add text "E" in the center (optional)
    try:
        # Try to use a font, fallback to default if not available
        font_size = 200
        try:
            font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", font_size)
        except:
            font = ImageFont.load_default()
        # Draw "E" text
        text = "E"
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
        position = ((512 - text_width) // 2, (512 - text_height) // 2 - 20)
        draw.text(position, text, fill=(255, 255, 255, 255), font=font)
    except Exception as e:
        print(f"Could not add text: {e}")

# Convert to RGBA if needed
if source_image.mode != 'RGBA':
    source_image = source_image.convert('RGBA')

print("\nGenerating PWA icons...")

for size in icon_sizes:
    # Resize image
    icon = source_image.resize((size, size), Image.Resampling.LANCZOS)

    # Save icon
    output_path = os.path.join(output_dir, f"icon-{size}x{size}.png")
    icon.save(output_path, 'PNG', optimize=True)
    print(f"✓ Created: icon-{size}x{size}.png")

print(f"\nDone! Icons generated in: {output_dir}")
print("You can now test PWA installation.")

File: generate-pwa-icons.php
<?php
/**
 * PWA Icon Generator
 * Generates all required PWA icons from favicon.ico
 *
 * Usage: php generate-pwa-icons.php
 */

$iconSizes = [72, 96, 128, 144, 152, 192, 384, 512];
$outputDir = __DIR__ . '/public/icons';
$faviconPath = __DIR__ . '/public/favicon.ico';

// Create icons directory if it doesn't exist
if (!file_exists($outputDir)) {
    mkdir($outputDir, 0755, true);
    echo "Created icons directory: $outputDir\n";
}

// Check if GD library is available
if (!extension_loaded('gd')) {
    echo "ERROR: GD library is not available. Please install php-gd extension.\n";
    echo "Alternatively, you can:\n";
    echo "1. Use online tools: https://www.pwabuilder.com/imageGenerator\n";
    echo "2. Install ImageMagick and use the convert command\n";
    echo "3. Manually create icons and place them in $outputDir\n";
    exit(1);
}

// Check if favicon exists
if (!file_exists($faviconPath)) {
    echo "ERROR: favicon.ico not found at $faviconPath\n";
    exit(1);
}

// Try to read favicon
$faviconData = file_get_contents($faviconPath);
if ($faviconData === false) {
    echo "ERROR: Could not read favicon.ico\n";
    exit(1);
}

// Try to create image from favicon
$sourceImage = @imagecreatefromstring($faviconData);
if ($sourceImage === false) {
    // If favicon is ICO format, try to create a simple colored icon
    echo "Note: Creating simple colored icons (favicon.ico format not directly supported by GD)\n";
    $sourceImage = imagecreatetruecolor(512, 512);
    $green = imagecolorallocate($sourceImage, 40, 167, 69); // #28a745
    $white = imagecolorallocate($sourceImage, 255, 255, 255);
    imagefill($sourceImage, 0, 0, $green);

    // Draw a simple leaf icon (E-Tinda logo)
    $centerX = 256;
    $centerY = 256;
    $radius = 200;

    // Draw a circle
    imagefilledellipse($sourceImage, $centerX, $centerY, $radius * 2, $radius * 2, $white);

    // Draw leaf shape (simplified)
    $points = [
        $centerX, $centerY - 100,
        $centerX + 80, $centerY - 40,
        $centerX + 60, $centerY + 40,
        $centerX, $centerY + 80,
        $centerX - 60, $centerY + 40,
        $centerX - 80, $centerY - 40,
    ];
    imagefilledpolygon($sourceImage, $points, 6, $green);
}

// Get original dimensions
$sourceWidth = imagesx($sourceImage);
$sourceHeight = imagesy($sourceImage);

echo "Generating PWA icons...\n";

foreach ($iconSizes as $size) {
    // Create new image with desired size
    $icon = imagecreatetruecolor($size, $size);

    // Enable alpha blending for transparency
    imagealphablending($icon, false);
    imagesavealpha($icon, true);

    // Fill with transparent background
    $transparent = imagecolorallocatealpha($icon, 0, 0, 0, 127);
    imagefill($icon, 0, 0, $transparent);

    // Resize and copy source image
    imagecopyresampled(
        $icon,
        $sourceImage,
        0, 0, 0, 0,
        $size, $size,
        $sourceWidth, $sourceHeight
    );

    // Save icon
    $outputPath = "$outputDir/icon-{$size}x{$size}.png";
    if (imagepng($icon, $outputPath, 9)) {
        echo "✓ Created: icon-{$size}x{$size}.png\n";
    } else {
        echo "✗ Failed: icon-{$size}x{$size}.png\n";
    }

    imagedestroy($icon);
}

imagedestroy($sourceImage);

echo "\nDone! Icons generated in: $outputDir\n";
echo "You can now test PWA installation.\n";


